parsl/tests/test_bash_apps/test_memoize_ignore_args_regr.py

```diff
@@ -30,21 +28,17 @@ def no_checkpoint_stdout_app(stdout=None):
     return "echo X"
 
 
-def test_memo_stdout():
-
+def test_memo_stdout(tmpd_cwd):
     assert const_list_x == const_list_x_arg
 
-    path_x = "test.memo.stdout.x"
-    if os.path.exists(path_x):
-        os.remove(path_x)
+    path_x = tmpd_cwd / "test.memo.stdout.x"
 
     # this should run and create a file named after path_x
-    no_checkpoint_stdout_app(stdout=path_x).result()
-    assert os.path.exists(path_x)
+    no_checkpoint_stdout_app(stdout=str(path_x)).result()
```

Comment:
Recent work *should* have this still work with path_x being an os.PathLike, which I think is the case here - for example, see the monitoring tests in parsl/tests/test_monitoring/test_stdouterr.py (although those tests only check that the path goes through to monitoring correctly - not that the path really is used by the executing app)

Reply:
(i.e. this `str` should be unnecessary)
(and maybe should be `os.fspath`)
(at which point you might get a `bytes`)
(and have to deal with that)